### ISSUE TITLE
embedding Citizen Reporter into Hub page

### ISSUE THREAD

**Issue by Nataliya1** [Esri/crowdsource-reporter#479]
hi @aparveen. I have a Citizen Reporter site with AGO signin enabled (only AGO signin, and no anonymous accss). And the sign in works as expected, but if I embed my Citizen Reporter app into a Hub page, I get an extra Signin window. I am not sure if that's the issue with Citizen Reporter or a Hub. 
![image](https://user-images.githubusercontent.com/6840324/97741382-a0d7de00-1ab0-11eb-9565-514e8e353106.png)
Please advise. Thank you.

**Comment by aparveen** [comment]
@Nataliya1 we will look into it.

**Comment by aparveen** [comment]
@Nataliya1 is your app shared publicly? Also, how are you embedding the app via iframe or the application card?

**Comment by Nataliya1** [comment]
@aparveen yes, it is shared publicly. I tried both iframe and an application card, and got the same outcome.

**Comment by chris-fox** [comment]
Hi @Nataliya1, this isn't specific to Hub. Unfortunately, this is the behavior with any app or web map embedded in an iframe and has to do with how the oauth login dialog is handled in an iframe for security reasons. You will notice if you click OK, it will open a new window with the typically sign-in dialog.

I do have a potential suggestion. Assuming you are doing this because the expectation is that users will create community accounts and to sign in with those accounts to log a report. One potential workaround would be to only share the problem reporter app with the Followers group. You can also now configure rows in the site to only share with users who are members of specific groups. 

So you could hide the app from the site and only make available to members of the followers group. Then instruct the user to sign in and follow, then they would have access to the app and already be signed into the site. If they come back in the future they will just have to sign in through the site before they can access the app, and they will never hit this issue.

**Comment by Nataliya1** [comment]
Hi @chris-fox thank you for the information. I will try your suggestion and report back.

**Comment by Nataliya1** [comment]
@chris-fox, I was able to test your workaround, and it seems to be working as expected.

But I do have another obstacle, which I posted about in here, issue #476 . 

So with your solution this is how my Hub page looks like with Embedded app, when a user doesn't have an account yet (hasn't join the initiative). And this is great. 
![image](https://user-images.githubusercontent.com/6840324/97926878-88262d00-1d29-11eb-896c-4769472fc860.png)

But once a user joins the initiative (creates a hub account), the citizen reporter app will become visible, but because there is a Signin/Splash window a user has to click on 'Sign in with' to get to the app itself. This an extra click, and I with there was a way to get rid of the Home/Splash screen, but still have option to sign in. Or a least have more options to customize the Home/Splash screen - issue #477. 

**Instead of this:** 
![image](https://user-images.githubusercontent.com/6840324/97927190-21edda00-1d2a-11eb-8375-ca6fe53046b8.png)

**Go right to this:**
![image](https://user-images.githubusercontent.com/6840324/97927327-68433900-1d2a-11eb-819e-97da872aa974.png)


Any suggestions? 

Appreciate your help a lot. 
Nataliya

**Comment by chris-fox** [comment]
@Nataliya1, so this is what I am finding. If the app isn't shared publicly, then when I access the app I am logged in (even when embedded in the Hub site). When I submit a new problem report or comment on an existing report, my user name is stored in the editor tracking. So it is actually logged into the app and storing the user id.

However, because the login button isn't available the My Submissions capability isn't available. Is the the primary concern you have as well?

**Comment by Nataliya1** [comment]
@chris-fox, Yes, that is part of the problem that people cannot track their submissions under 'My Submissions' dropdown, when the Home/Splash screen is removed. We really would like to have that capability.
